src/main/java/sirius/db/mixing/properties/MultiLanguageStringProperty.java

```diff
@@ -97,6 +99,30 @@ protected void onBeforeSaveChecks(Object entity) {
                                                    .handle();
                                });
         }
+
+        Object propertyValue = getValue(entity);
+        boolean withFallback = multiLanguageString.isWithFallback();
+        Map<String, String> values = ((Map<String, String>) propertyValue);
+
+        if (!isNullable()) {
+            if(withFallback && Strings.isEmpty(values.get(MultiLanguageString.FALLBACK_KEY))) {
+                throw Exceptions.createHandled()
+                                .error(new InvalidFieldException(getName()))
+                                .withNLSKey("MultiLanguageStringProperty.fallbackNotSet")
+                                .set("field", getFullLabel())
+                                .handle();
+            }
+
+            boolean consideredEmpty = isConsideredNull(propertyValue);
```

Comment:
shoudln't we replace this with Strings.isEmpty as it is way clearer what happens here. Also I#d use pretty much a stream pipeline as defined above